### PR TITLE
Dynamic Events on Homepage

### LINKED ISSUE #52: Events tool

### LINKED ISSUE #77: Accept long urls for event fliers and links

### LINKED ISSUE #80: Add backend deploy preview
We are currently pushing the `src/generated/graphql.tsx` file because we cannot generate the most up-to-date generated file during deployment. Add a 'backend deploy preview' similar to netlify so that we don't need to pushn `src/generated/graphql.tsx`

### LINKED ISSUE #103: Modify homepage to use hooks + backend for events

### LINKED ISSUE #113: Upload images to azure storage and remove all images from netlify

### LINKED ISSUE #127: Graphql query for current events

### PR DESCRIPTION
**Pull request by ClayMav**
# Pull Request Description
Closes #103 
Closes #77 
Closes #127 
Closes #113 
Closes #80 
Closes #52
Fetches events from graphql and displays, few issues:
- hardcoded sig logo on the frontend #106 #107 
- Several re-renders on load, I think due to way that filtering is done #108 

TODO

- [x] Fix filtering

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by ClayMav** [comment]
Closes #127

**Comment by ClayMav** [comment]
Closes #113

**Comment by ClayMav** [comment]
Closes #80

**Comment by ClayMav** [comment]
Closes #52

File: apps/web/src/screens/HomePage/sections/Events/Event.tsx [hunk 1]
```diff
@@ -173,91 +176,130 @@ const Description = styled.div`
...
             ellipsis="... read more"
             basedOn="words"
             style={{ fontSize: "16px" }}
           />
         </Description>
         <FlierLink
-          href={require("../../../../static/img/" +
-            (doesFlierExist(event.img_path) ? event.img_path : "acm.png"))}
+          href={
+            event.flierLink
```
Comment (kevinschoonover) [inline apps/web/src/screens/HomePage/sections/Events/Event.tsx]: See above for variable comment

File: apps/web/src/screens/HomePage/sections/Events/index.tsx [hunk 2]
```diff
@@ -145,38 +171,60 @@ const CalendarLink = styled.a`
...
 ];
 
 const DEFAULT_EVENTS_TO_DISPLAY: number = 3;
 const CALENDAR_LINK: string =
   "https://calendar.google.com/calendar/embed?src=mst.edu_7u3stm8bn7l2umuastep5fmbl0%40group.calendar.google.com&ctz=America%2FChicago";
 
 const Events: React.FC<{}> = (): JSX.Element => {
-  const [filters, setFilters] = useState<boolean[]>([]);
+  const result: GetCurrentEventsQueryHookResult = useGetCurrentEventsQuery();
+  let events: any;
```
Comment (kevinschoonover) [inline apps/web/src/screens/HomePage/sections/Events/index.tsx]: Should use the `Event[]` type exported from the generated graphql

File: apps/web/src/screens/HomePage/sections/Events/index.tsx [hunk 3]
```diff
@@ -145,38 +171,60 @@ const CalendarLink = styled.a`
...
+  if (result.data && result.data.currentEvents) {
+    events = result.data.currentEvents;
+    events.sort(
+      (a: any, b: any) =>
```
Comment (kevinschoonover) [inline apps/web/src/screens/HomePage/sections/Events/index.tsx]: Use the exported `Event` type from generated graphql

File: apps/web/src/screens/HomePage/sections/Events/index.tsx [hunk 4]
```diff
@@ -145,38 +171,60 @@ const CalendarLink = styled.a`
...
+  "Competition",
+  "Data",
+  "Game",
+  "General",
+  "Hack",
+  "Security",
+  "Women"
 ];
 
 const DEFAULT_EVENTS_TO_DISPLAY: number = 3;
```
Comment (kevinschoonover) [inline apps/web/src/screens/HomePage/sections/Events/index.tsx]: outside of the scope of your PR, but is this something we want to put in the config?

File: apps/web/src/screens/HomePage/sections/Events/index.tsx [hunk 5]
```diff
@@ -145,38 +171,60 @@ const CalendarLink = styled.a`
...
+  let events: any;
+  if (result.data && result.data.currentEvents) {
+    events = result.data.currentEvents;
+    events.sort(
+      (a: any, b: any) =>
+        new Date(a.dateHosted).getTime() - new Date(b.dateHosted).getTime()
+    );
+  }
+
+  const [filters, setFilters] = useState<boolean[]>([
```
Comment (kevinschoonover) [inline apps/web/src/screens/HomePage/sections/Events/index.tsx]: Try `new Array(FILTER_TYPES.length).fill(false);`. We can then generate this list via graphql dynamically later

File: apps/web/src/screens/HomePage/sections/Events/index.tsx [hunk 6]
```diff
@@ -233,27 +281,13 @@ const Events: React.FC<{}> = (): JSX.Element => {
...
-                  <ImgImg
-                    src={require("../../../../static/img/hack-dark.png")}
-                  />
-                  <ImgImg
-                    src={require("../../../../static/img/sec-dark.png")}
-                  />
-                  <ImgImg
-                    src={require("../../../../static/img/acm-w-dark.png")}
-                  />
+                  <ImgImg src={`${config.CDN_URI}/static/comp-dark.png`} />
```
Comment (kevinschoonover) [inline apps/web/src/screens/HomePage/sections/Events/index.tsx]: :heart: :heart: :heart: :heart:

File: apps/web/src/screens/HomePage/sections/Events/index.tsx [hunk 7]
```diff
@@ -311,12 +345,12 @@ const Events: React.FC<{}> = (): JSX.Element => {
               </Sigs>
             </FilterWrapper>
             <EventsWrapper>
-              {eventsData.events
-                .filter(event => {
-                  return showEvent(event.group);
+              {events
+                .filter((event: any) => {
```
Comment (kevinschoonover) [inline apps/web/src/screens/HomePage/sections/Events/index.tsx]: Use exported `Event` type from graphql

File: apps/web/src/screens/HomePage/sections/Events/index.tsx [hunk 8]
```diff
@@ -311,12 +345,12 @@ const Events: React.FC<{}> = (): JSX.Element => {
...
+                  return showEvent(event.hostSig.name);
                 })
                 .slice(0, maxEvents)
-                .map((event, i) => {
+                .map((event: any, i: number) => {
```
Comment (kevinschoonover) [inline apps/web/src/screens/HomePage/sections/Events/index.tsx]: Use exported `Event` type from graphql

**Comment by kevinschoonover** [comment]
Also, you gotta make migrations for these backend changes

File: apps/admin-web/src/App.tsx [hunk 9]
```diff
@@ -89,8 +92,8 @@ const App: React.SFC<{}> = (): JSX.Element => {
           </Header>
           <Content style={{ margin: "24px 16px 0" }}>
             <MainContent />
+            <Footer style={{ textAlign: "center" }}>S&T ACM 2019</Footer>
```
[the remark this answers is not in the document]
Comment (ClayMav) [reply apps/admin-web/src/App.tsx]: ye

[the remark this answers is not in the document]
Comment (kevinschoonover) [reply apps/admin-web/src/App.tsx]: https://ant.design/components/layout/ based on what im seeing here the footer is always outside of content

**Pull request merged**